armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -64,26 +70,25 @@ def expandColdDimsToHot(assems, isDetailedAxialExpansion, referenceAssembly=None
     assems: list[:py:class:`Assembly <armi.reactor.assemblies.Assembly>`]
         list of assemblies to be thermally expanded
     isDetailedAxialExpansion: bool
-        If true assemblies will be forced to conform to the reference mesh after expansion
+        If False, assemblies will be forced to conform to the reference mesh after expansion
+    assemsToSkip: List[str]
+        list of strings to be converted to flags to indicate which assemblies to skip axial expansion
     referenceAssembly: :py:class:`Assembly <armi.reactor.assemblies.Assembly>`, optional
-        Assembly whose mesh other meshes wil conform to if isDetailedAxialExpansion is true.
+        Assembly whose mesh other meshes will conform to if isDetailedAxialExpansion is False.
         If not provided, will assume the finest mesh assembly which is typically fuel.
     """
+    aToSkip = list(Flags.fromStringIgnoreErrors(t) for t in assemsToSkip)
     assems = list(assems)
     if not referenceAssembly:
         referenceAssembly = getDefaultReferenceAssem(assems)
     axialExpChanger = AxialExpansionChanger(isDetailedAxialExpansion)
     for a in assems:
-        if not a.hasFlags(Flags.CONTROL):
-            axialExpChanger.setAssembly(a)
-            # this doesn't get applied to control assems, so CR will be interpreted
-            # as hot. This should be conservative because the control rods will
-            # be modeled as slightly shorter with the correct hot density. Density
-            # is more important than height, so we are forcing density to be correct
-            # since we can't do axial expansion (yet)
-            axialExpChanger.applyColdHeightMassIncrease()
-            axialExpChanger.expansionData.computeThermalExpansionFactors()
-            axialExpChanger.axiallyExpandAssembly()
+        if a.hasFlags(Flags.CONTROL) or any(a.hasFlags(aFlags) for aFlags in aToSkip):
+            continue
```

Comment:
Assuming it's okay to require exact type IDs here, e.g., `"duct"` wouldn't match `"innerduct"`, you could just do the following:

```suggestion
        if a.hasFlags(Flags.CONTROL) or a.hasFlags(aToSkip, exact=True):
            continue
```

Reply:
My concern with using exact if you have the following:

```python
import armi
armi.configure()
from armi.reactor.blocks import HexBlock

b0 = HexBlock("fuel")
b0.setType("fuel")
b1 = HexBlock("outer fuel")
b1.setType("outer fuel")

# Part 1: Use exact to get false negative
print(b1.hasFlags(typeID=[b0.p.flags]))
print(b1.hasFlags(typeID=[b0.p.flags], exact = True))

# Part 2: Gives false negative always
print(b0.hasFlags(typeID=[b1.p.flags]))
print(b0.hasFlags(typeID=[b1.p.flags], exact = True))

# Part 3: The correction for part 2
print(any(b0.hasFlags(f) for f in b1.p.flags))
print(any(b0.hasFlags(f, exact=True) for f in b1.p.flags))
```
which prints
```
True
False
False
False
True
True
```
So using the `any` approach is a bit more robust, I think. It's also used in Core::processLoading here, https://github.com/terrapower/armi/blob/c8a40c76d9da31e3f8322a51eb52d3864a3f68a6/armi/reactor/reactors.py#L2272-L2276